core/src/actions/destinations.ts

```diff
@@ -251,46 +258,46 @@ export class DestinationConnectionOptions extends AuthenticatedAction {
 }
 
 export class DestinationMappingOptions extends AuthenticatedAction {
-  constructor() {
-    super();
-    this.name = "destination:mappingOptions";
-    this.description = "return option choices from this destination mapping";
-    this.outputExample = {};
-    this.permission = { topic: "destination", mode: "read" };
-    this.inputs = {
-      id: { required: true },
-    };
-  }
-
-  async runWithinTransaction({ params }) {
+  name = "destination:mappingOptions";
+  description = "return option choices from this destination mapping";
+  outputExample = {};
+  permission: ActionPermission = { topic: "destination", mode: "read" };
+  inputs = {
+    id: { required: true },
+  };
+
+  async runWithinTransaction({
+    params,
+  }: {
+    params: ParamsFrom<DestinationMappingOptions>;
+  }) {
     const destination = await Destination.findById(params.id);
     const options = await destination.destinationMappingOptions(false); // never use cache when displaying to the user
 
-    const _destinationTypeConversions: { [key: string]: Array<string> } = {};
-    for (const k in destinationTypeConversions) {
-      _destinationTypeConversions[k] = Object.keys(
-        destinationTypeConversions[k]
-      );
+    const _destinationTypeConversions: Record<string, any> = {};
+    for (const [k, v] of Object.entries(destinationTypeConversions)) {
+      _destinationTypeConversions[k] = Object.keys(v);
     }
```

Comment:
Using `Object.entries` to get the keys and values of an object provides better typing